configuration/configuration.go

```diff
@@ -143,6 +146,10 @@ func (c *Data) setConfigDefaults() {
 	c.v.SetDefault(varClientSecret, "tenantsecretNew")
 }
 
+func (c *Data) Set(key string, value interface{}) {
+	c.v.Set(key, value)
```

Comment:
Oh.. I actually missed it in the other PR... So, you need to export Set() method in Configuration to be able to use it in tests. But it's error prone because Set() can be used in tests only. It may cause issues if used in runtime code because Set() is not thread safe in viper...

So, you have to use env vars :( or as alternative you could use some configuration interfaces instead of passing the whole configuration.Data.
For example something like this: https://github.com/fabric8-services/fabric8-auth/blob/master/account/service/tenant.go#L26
https://github.com/fabric8-services/fabric8-auth/blob/master/account/service/tenant.go#L36
https://github.com/fabric8-services/fabric8-auth/blob/master/account/service/tenant_whitebox_test.go#L37

Reply:
solved in 5136ba5
the same approach (the same helper functions) I'm [using](cb77388) in the other PR, so merging should be easier.